collector/pkg/component/analyzer/cpuanalyzer/cpu_analyzer.go

```diff
@@ -90,12 +97,53 @@ func (ca *CpuAnalyzer) ConsumeEvent(event *model.KindlingEvent) error {
 func (ca *CpuAnalyzer) ConsumeTransactionIdEvent(event *model.KindlingEvent) {
 	isEntry, _ := strconv.ParseUint(event.GetStringUserAttribute("is_enter"), 10, 32)
 	ev := &TransactionIdEvent{
-		Timestamp: event.Timestamp,
-		TraceId:   event.GetStringUserAttribute("trace_id"),
-		IsEntry:   uint32(isEntry),
+		Timestamp:   event.Timestamp,
+		TraceId:     event.GetStringUserAttribute("trace_id"),
+		IsEntry:     uint32(isEntry),
+		Protocol:    event.GetStringUserAttribute("protocol"),
+		Url:         event.GetStringUserAttribute("url"),
+		PidString:   strconv.FormatUint(uint64(event.GetPid()), 10),
+		ContainerId: event.GetContainerId(),
 	}
 	//ca.sendEventDirectly(event.GetPid(), event.Ctx.ThreadInfo.GetTid(), event.Ctx.ThreadInfo.Comm, ev)
 	ca.PutEventToSegments(event.GetPid(), event.Ctx.ThreadInfo.GetTid(), event.Ctx.ThreadInfo.Comm, ev)
+	ca.analyzerJavaTraceTime(ev)
+}
+
+func (ca *CpuAnalyzer) analyzerJavaTraceTime(ev *TransactionIdEvent) {
+	if ev.IsEntry == 1 {
+		ca.javaTraces[ev.TraceId+ev.PidString] = ev
+	} else {
+		pid, _ := strconv.ParseInt(ev.PidString, 10, 64)
+		spendTime := ev.Timestamp - ca.javaTraces[ev.TraceId+ev.PidString].Timestamp
```

Comment:
Should assign `ca.javaTraces[ev.TraceId+ev.PidString]` to a new variable to reduce the query to the map in the subsequence code.